src/back-end/lib/resources/proposal/code-with-us.ts

```diff
@@ -100,7 +100,7 @@ const resource: Resource = {
       if (!await permissions.readManyCWUProposals(connection, request.session, request.query.opportunity)) {
         return respond(401, [permissions.ERROR_MESSAGE]);
       }
-      const dbResult = await db.readManyCWUProposals(connection, request.query.opportunity);
+      const dbResult = await db.readManyCWUProposals(connection, request.session, request.query.opportunity);
```

Comment:
You can use `permissions.isSignedIn(request.session)` for the type system to know whether the session is an `AuthenticatedSession` or not.